connection.go

```diff
@@ -90,21 +90,37 @@ func (c *conn) IsValid() bool {
 func (c *conn) ExecContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Result, error) {
 	log := logger.WithContext(c.id, driverctx.CorrelationIdFromContext(ctx), "")
 	msg, start := logger.Track("ExecContext")
+	defer log.Duration(msg, start)
+
 	ctx = driverctx.NewContextWithConnId(ctx, c.id)
 	if len(args) > 0 {
 		return nil, errors.New(ErrParametersNotSupported)
 	}
 	exStmtResp, opStatusResp, err := c.runQuery(ctx, query, args)
 
 	if exStmtResp != nil && exStmtResp.OperationHandle != nil {
+		// we have an operation id so update the logger
 		log = logger.WithContext(c.id, driverctx.CorrelationIdFromContext(ctx), client.SprintGuid(exStmtResp.OperationHandle.OperationId.GUID))
+
+		// since we have an operation handle we can close the operation
+		if opStatusResp == nil || opStatusResp.GetOperationState() != cli_service.TOperationState_CLOSED_STATE {
```

Comment:
is this enough? I think we also have to check `directResult.CloseOperationResp`, no?

Reply:
You're right. I tried it and the returned operation state can be 'Finished' (i.e. not closed) while the direct results have a TCloseOperationResp with a status of 'Success', indicating that the operation is in a closed state.